docarray/document/mixins/sugar.py

```diff
@@ -57,6 +57,34 @@ def match(
         ...
 
     def match(self: 'T', *args, **kwargs) -> 'T':
+        """Matching the current Document against a set of Documents.
```

Comment:
This is a fundamental problem that we have in our code structure. If we don't put it here then it will not be rendered in the docs, and be more or less useless to our users.
In the future we should avoid having only `*args` and `**kwargs` everywhere (why do we do this anyways), imo.

Until then we have two options:
1. have it in this sub-optimal state, like I'm doing here
2. write a CI bot that copies the docstring from the `@overload` stub to the "real" signature, like I did in core

Reply:
if it was done in core, why don't we do it here?